trunner/runners/common.py

```diff
@@ -308,21 +310,46 @@ def go(self):
 class Runner:
     """Common interface for test runners"""
 
+    @abstractmethod
+    def status(self, status):
```

Comment:
For `status` please use class consts instead of strings You can easily mistype (check how it's done in `TestCase` class).

I'm missing "idle" status + initialization to IDLE in the constructor here.

If we use `@abstractmethod` properly  - qemu runners will fail as this method is not overriden there (so think carefully if it really should be abstract here)

Reply:
Thanks, changed to consts

Because of using only 3 led collors, there are 3 states: busy, pass, fail. The idle state is pass or fail (depend on last runner run). In the future there can be more statuses

I've put it here, because I was afraid that without this declaration runners without status method can't be called from self.runners[target].status(). But that's right, if it's abstract, they should have it defined. Can I leave it here without @abstractmethod or add empty definitions for Qemu, Host and STM Runner?

Reply:
- I proposed "idle" as it's a good value for initialization. In this particular case maybe "busy" is a good init value, but we should write in comment why is that so (as it seems counter-intuitive that the runner status is busy when it's not running any test, this is actually the "DUT status" (device under test)
- maybe the default implementation shouldn't be empty, but should at least set `self.status` to be available for getting in the future?
- the mapping 1-1 status to gpio is the most basic one (but well enough for now), I can think of having one RGB led where You can mix any color You want using 3 PWM channels in the future :)

Reply:
Sorry, I missed that, so I needed another force push to apply it

- You are right, I've added additional comment

- I agree that's better, applied

- Yes, that is an RGB diode and mixing colors using PWM was the plan, but because it's not so important there is simpler solution now